FIX: Restructure Item Delegates

zdomke opened the pull request:
Restructure ItemDelegates to make an abstract class to manage the editor_list. Added functionality to reset the editor_list on model reset.

YektaY commented on archive_viewer/widgets/item_delegates.py:
```diff
@@ -5,11 +5,42 @@
+        model.modelAboutToBeReset.connect(self.reset_editors)
+
+    def paint(self, painter: QPainter, option: QStyleOptionViewItem, index: QModelIndex) -> None:
+        """Create a new persistent editor on the Table View at the given index."""
```
can you add the return and parameters to the doc for the methods added by this PR? Thanks

zdomke replied:
I updated the docstrings and return parameters

YektaY reviewed (commented):
This looks good! Would just request that the doc stings have the parameters and returns added to them.

zdomke commented:
Additional changes:
1. Added abstractmethods to EditorDelegate
2. Removed the unused delegates SliderDelegate and CheckboxDelegate
  a. This also included removing the now unused widget CenterCheckbox (deprecated in #16)
